instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/TracingRequestApiGatewayWrapper.java

```diff
@@ -5,23 +5,58 @@
 
 package io.opentelemetry.instrumentation.awslambda.v1_0;
 
+import com.amazonaws.services.lambda.runtime.Context;
 import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
 import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.fasterxml.jackson.core.JsonProcessingException;
 import io.opentelemetry.sdk.OpenTelemetrySdk;
+import java.util.function.BiFunction;
 
 /**
  * Wrapper for {@link TracingRequestHandler}. Allows for wrapping a lambda proxied through API
  * Gateway, enabling single span tracing and HTTP context propagation.
  */
 public class TracingRequestApiGatewayWrapper
-    extends TracingRequestWrapperBase<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    extends TracingRequestWrapperBase<APIGatewayProxyRequestEvent, Object> {
 
   public TracingRequestApiGatewayWrapper() {
-    super();
+    super(TracingRequestApiGatewayWrapper::map);
   }
 
   // Visible for testing
-  TracingRequestApiGatewayWrapper(OpenTelemetrySdk openTelemetrySdk, WrappedLambda wrappedLambda) {
-    super(openTelemetrySdk, wrappedLambda);
+  TracingRequestApiGatewayWrapper(
+      OpenTelemetrySdk openTelemetrySdk,
+      WrappedLambda wrappedLambda,
+      BiFunction<APIGatewayProxyRequestEvent, Class, Object> mapper) {
+    super(openTelemetrySdk, wrappedLambda, mapper);
+  }
+
+  // Visible for testing
+  static Object map(APIGatewayProxyRequestEvent event, Class clazz) {
+    try {
+      return OBJECT_MAPPER.readValue(event.getBody(), clazz);
+    } catch (JsonProcessingException e) {
+      throw new IllegalStateException(
+          "Could not map API Gateway event body to requested parameter type: " + clazz, e);
+    }
+  }
+
+  @Override
+  protected APIGatewayProxyResponseEvent doHandleRequest(
+      APIGatewayProxyRequestEvent input, Context context) {
+    Object result = super.doHandleRequest(input, context);
+    APIGatewayProxyResponseEvent event = null;
+    // map to response event if needed
+    if (result instanceof APIGatewayProxyResponseEvent) {
+      event = (APIGatewayProxyResponseEvent) result;
+    } else {
+      try {
+        event = new APIGatewayProxyResponseEvent();
+        event.setBody(OBJECT_MAPPER.writeValueAsString(result));
```

Comment:
Do we know that our OBJECT_MAPPER is configured the same as Lambda?

Reply:
Unfortunately no guarantees here. Haven't seen an official spec by AWS guys regarding how the JSON is mapped in their code.